### PR TITLE
Handle missing Nami collateral

### LINKED ISSUE #305: Handle missing Nami collateral
At the moment, connecting to a Nami wallet that has not set its special collateral can lead to difficult-to-track bugs. We need to handle this explicitly, either in `namiUtxosAt` where we attempt to filter the collateral out (necessary if collateral is set, otherwise Nami will freak out during balancing/submission) or during wallet initialization (perhaps we should throw an exception?)

### PR DESCRIPTION
**Pull request by jy14898**
Closes #305

### CODE REVIEW

File: src/QueryM/Utxos.purs [hunk 1]
```diff
@@ -89,13 +91,12 @@ utxosAt addr = asks _.wallet >>= maybe (allUtxosAt addr) (utxosAtByWallet addr)
...
   namiUtxosAt :: Address -> QueryM (Maybe Transaction.UtxoM)
-  namiUtxosAt address = do
-    utxos' <- allUtxosAt address
-    collateral' <- getWalletCollateral
-    pure do
-      utxos <- unwrap <$> utxos'
-      collateral <- unwrap <$> collateral'
-      pure $ wrap $ Map.delete collateral.input utxos
+  namiUtxosAt address = getWalletCollateral >>= maybe
+    (liftEffect $ throw "Nami wallet missing collateral")
```
Comment (jy14898) [inline src/QueryM/Utxos.purs:95]: I'm throwing the error in Aff here, but above it suggests we should move to Either for fine-grained error handling. Although, it seems the errors inside the Maybe are only a small subset of the possible errors from calling Ogmios?

**Comment by jy14898** [comment]
I haven't added the check during wallet initialisation as I think it's somewhat superfluous to have (at least without the changes I've made thus far). If a wallet lost their collateral during a previous transaction or removed it themselves, we'd be back to square one with difficult to solve bugs.

It may still be worth adding however just to fail faster

File: src/QueryM/Utxos.purs (hunk 1 above)
Comment (ngua) [reply src/QueryM/Utxos.purs:95]: I think throwing an exception is fine. This isn't an error that can be meaningfully recovered from. Nami doesn't really work without collateral set

**Comment by ngua** [comment]
I think we should also add a check during wallet initialization

File: src/QueryM/Utxos.purs (hunk 1 above)
Comment (ngua) [reply src/QueryM/Utxos.purs:95]: That said, we can also change the return type of `utxosAt` to `Either ...`

**Comment by jy14898** [comment]
@ngua I've added the check at wallet init.

Would you like me to make the changes to `utxosAt` to move to using `Either`?

**Comment by jy14898** [comment]
FYI I've yet to test this PR on a real wallet as I'm waiting for cardano-node to finish syncing. Should be done soon

**Comment by ngua** [comment]
I can try running the example with Nami

**Comment by jy14898** [comment]
> I can try running the example with Nami

If you'd like, but I think it's about to finish in the next couple of minutes. Currently at block 57264210 of 57376972

**Comment by ngua** [comment]
OK, I ran it with my main account with collateral as well as a new account without, seems to be working as intended

**Review by ngua — APPROVED** [review]

**Pull request merged**